src/Microsoft.ApplicationInsights/Extensibility/Implementation/OperationHolder.cs

```diff
@@ -80,18 +80,26 @@ protected virtual void Dispose(bool disposing)
                             var currentActivity = Activity.Current;
                             if (currentActivity == null || operationTelemetry.Id != currentActivity.Id)
                             {
-                                // this is for internal error reporting
-                                CoreEventSource.Log.InvalidOperationToStopError();
-
-                                // this are details with unique ids for debugging
-                                CoreEventSource.Log.InvalidOperationToStopDetails(
-                                    string.Format(
-                                        CultureInfo.InvariantCulture,
-                                        "Telemetry Id '{0}' does not match current Activity '{1}'",
-                                        operationTelemetry.Id,
-                                        currentActivity?.Id));
-
-                                return;
+                                // W3COperationCorrelationTelemetryInitializer changes Id
+                                // but keeps an original one in 'ai_legacyRequestId' property
+
+                                if (!operationTelemetry.Properties.TryGetValue("ai_legacyRequestId", out var legacyId) ||
+                                    legacyId != currentActivity?.Id)
+                                {
+                                    // this is for internal error reporting
+                                    CoreEventSource.Log.InvalidOperationToStopError();
+
+                                    // this are details with unique ids for debugging
+                                    CoreEventSource.Log.InvalidOperationToStopDetails(
```

Comment:
This message does not go into the default diagnostic traces, right? It seems to be "Warning", but not "Info"/"Verbose". Still not "Error" but enough to raise a question on it's relative severity.

Reply:
nothing new here too. There are two messages: 
1. AI Internal error that says that ids do not match - this shows up in users telemetry and our telemetry 
2. details that include ids - this is warning for debugging with perfview